app/models/target_remote_execution_proxy.rb

```diff
@@ -0,0 +1,4 @@
+class TargetRemoteExecutionProxy < ActiveRecord::Base
+  belongs_to :remote_execution_proxy, :class_name => 'SmartProxy'
+  belongs_to :target, :polymorphic => true
```

Comment:
It's polymorphic to enable association directly to host? Just checking, if I understand the intention properlỵ...